Tangem/Modules/ExpressView/ExpressViewModel.swift

```diff
@@ -646,121 +582,38 @@ private extension ExpressViewModel {
-        }
-    }
-
-    func updateRequiredPermission(isPermissionRequired: Bool) {
-        if isPermissionRequired {
-            let symbol = interactor.getSender().tokenItem.blockchain.currencySymbol
-            permissionInfoRowViewModel = DefaultWarningRowViewModel(
-                title: Localization.swappingGivePermission,
-                subtitle: Localization.swappingPermissionSubheader(symbol),
-                leftView: .icon(Assets.swapLock)
-            )
-        } else {
-            permissionInfoRowViewModel = nil
-        }
-    }
+extension ExpressViewModel: NotificationTapDelegate {
+    func didTapNotification(with id: NotificationViewId) {}
 
-    func updatePendingApprovingTransaction(hasPendingTransaction: Bool) {
-        if hasPendingTransaction {
-            pendingTransaction = DefaultWarningRowViewModel(
-                title: Localization.swappingPendingTransactionTitle,
-                subtitle: Localization.swappingPendingTransactionSubtitle,
-                leftView: .loader
-            )
-        } else {
-            pendingTransaction = nil
+    func didTapNotificationButton(with id: NotificationViewId, action: NotificationButtonActionType) {
+        guard
+            let notif = notificationInputs.first(where: { $0.id == id }),
+            notif.settings.event is ExpressNotificationEvent
+        else {
+            return
         }
-    }
 
-    func updateEnoughAmountForFee(isNotEnoughAmountForFee: Bool) {
-        if isNotEnoughAmountForFee {
-            let symbol = interactor.getSender().tokenItem.blockchain.currencySymbol
-            feeWarningRowViewModel = DefaultWarningRowViewModel(
-                subtitle: Localization.swappingNotEnoughFundsForFee(symbol, symbol),
-                leftView: .icon(Assets.attention)
-            )
-        } else {
-            feeWarningRowViewModel = nil
+        switch action {
+        case .refresh:
+            didTapWaringRefresh()
+        case .openNetworkCurrency:
+            openNetworkCurrency()
+        default: return
         }
     }
 
-    func updateRefreshWarningRowViewModel(message: String?) {
-        if let message {
-            refreshWarningRowViewModel = DefaultWarningRowViewModel(
-                subtitle: Localization.swappingErrorWrapper(message.capitalizingFirstLetter()),
-                leftView: .icon(Assets.attention),
-                rightView: .icon(Assets.refreshWarningIcon)
-            ) { [weak self] in
-                self?.didTapWaringRefresh()
-            }
-        } else {
-            refreshWarningRowViewModel = nil
+    private func openNetworkCurrency() {
+        guard
+            case .token(_, let blockchain) = initialWallet.tokenItem,
+            let networkCurrencyWalletModel = userWalletModel.walletModelsManager.walletModels.first(where: {
+                $0.tokenItem == .blockchain(blockchain) && $0.blockchainNetwork == initialWallet.blockchainNetwork
+            })
```

Comment:
```suggestion
            let networkCurrencyWalletModel = userWalletModel.walletModelsManager.walletModels.first(where: {
                $0.tokenItem == .blockchain(initialWallet.tokenItem.blockchain) && $0.blockchainNetwork == initialWallet.blockchainNetwork
            })
```

тут можно прям из `initialWallet.tokenItem` `blockchain` взять

Reply:
да, согласен, изменил